[added] description of nats based account resolver (#220)
nats-io/nats.docs: nats-server/configuration/securing_nats/jwt/resolver.md

> # Account lookup using Resolver

Reviewer: ```suggestion
# Account Lookup Using a Resolver
```

The `resolver` configuration option is used in conjunction with [NATS JWT Authentication](./) and [nsc](../../../../nats-tools/nsc/). The `resolver` option specifies a URL where the nats-server can retrieve an account JWT. There are three built-in resolver implementations:

* [`URL`](resolver.md#URL-Resolver)
* [`MEMORY`](resolver.md#Memory)

> * [nats based resolver](resolver.md#nats-based-resolver)

Reviewer: Is this the literal string specified?

Author: no it is not, we use a struct instead

For more information on how to configure a memory resolver, see [this tutorial](mem_resolver.md).

> ## nats based resolver

Reviewer: Capitalize NATS

Author: fixed

Author: fixed

NATS based resolver embed the functionality of the [account server](https://github.com/nats-io/nats-account-server) inside the nats-server.
To not have to store all account jwt on every server, this resolver has two sub types `full` and `cache`.
Their commonalities are that they exchange/lookup account jwt via NATS and the system account and store them in a local (not shared) directory.

> ### full

Reviewer: ```suggestion
### Full
```

Author: fixed

[`nsc`](../../../../nats-tools/nsc/README.md) supports push/pull/purge with this resolver type. 
[JWTs](../../nats-server/configuration/securing_nats/jwt/), uploaded this way, are stored in a directory the server has exclusive access to. 

```yaml
resolver: {
    type: full
    # Directory in which account jwt will be stored
    dir: './jwt'
    # In order to support jwt deletion, set to true
    # If the resolver type is full delete will rename the jwt.
    # This is to allow manual restoration in case of inadvertent deletion.
    # To restore a jwt, remove the added suffix .delete and restart or send a reload signal.
    # To free up storage you must manually delete files with the suffix .delete.
    allow_delete: false
    # Interval at which a nats-server with a nats based account resolver will compare
    # it's state with one random nats based account resolver in the cluster and if needed,
    # exchange jwt and converge on the same set of jwt.
    interval: "2m"
    # limit on the number of jwt stored, will reject new jwt once limit is hit.
    limit: 1000
}
```

> This resolver type also supports `resolver_preload`. When present jwt listed are stored in the resolver.

Reviewer: ```suggestion
This resolver type also supports `resolver_preload`. When present JWT's are listed are stored in the resolver.
```

There, they may be subject to updates. Restarts of the `nats-server` will hold on to these more recent versions.

Not every server in a cluster needs to be set to `full`.
You need enough to still serve your workload adequately, while some server are offline.

> ### cache

Reviewer: ```suggestion
### Cache
```

> This resolver only stores a subset of [JWT](../../nats-server/configuration/securing_nats/jwt/) and evicts extra ones based on an LRU scheme. 

Reviewer: ```suggestion
This resolver only stores a subset of [JWTs](../../nats-server/configuration/securing_nats/jwt/) and evicts others based on an LRU scheme. 
```

Missing jwt are downloaded from `full` nats based resolver. 
This resolver is essentially the URL Resolver in nats.

```yaml
resolver: {
    type: cache
    # Directory in which account jwt will be store
    dir: "./"
    # limit on the number of jwt stored, will evict old jwt once limit is hit.
    limit: 1000
    # How long to hold on to a jwt before discarding it. 
    ttl: "2m"
}
```

> ### nats based resolver - integration

Reviewer: ```suggestion
### NATS Based Resolver - Integration
```

> nats based resolver utilize the system account for lookup and upload of account [JWTs](../../nats-server/configuration/securing_nats/jwt/) .

Reviewer: ```suggestion
The NATS based resolver utilizes the system account for lookup and upload of account [JWTs](../../nats-server/configuration/securing_nats/jwt/) .
```

If your application requires tighter integration you can make use of these subjects for tighter integration.

> To upload or update a possibly on the fly generated account jwt without [`nsc`](../../../../nats-tools/nsc/README.md), send it as request to `$SYS.REQ.CLAIMS.UPDATE`.

Reviewer: ```suggestion
To upload or update any generated account jwt without [`nsc`](../../../../nats-tools/nsc/README.md), send it as request to `$SYS.REQ.CLAIMS.UPDATE`.
```
